src/tm_devices/drivers/pi/scopes/tekscope_2k/tekscope_2k.py

```diff
@@ -18,10 +24,117 @@ class TekScope2k(Scope, ABC):
     # Properties
     ################################################################################################
 
+    @cached_property
+    def total_channels(self) -> int:
+        """Return the total number of channels (all types)."""
+        return int(self.model[6])
+
     ################################################################################################
     # Public Methods
     ################################################################################################
 
+    def waveforms_displayed(self) -> List[str]:
+        """Returns list of waveforms currently displayed on oscilloscope screen.
```

Comment:
```suggestion
        """Get the list of waveforms currently displayed on the oscilloscope screen and available as data sources.
```

Reply:
It seems like there was an issue with the comment being too long and extending to second line. Maybe I wasn't doing something right, but I shortened the comment to fit on one line.